libgit/autogit_manager.go

```diff
@@ -119,6 +130,148 @@ func (am *AutogitManager) getNewConfigDefault(ctx context.Context) (
 	return getNewConfig(ctx, am.config, am.kbCtx, am.kbfsInitParams, am.log)
 }
 
+// commonTime computes the current time according to our estimate of
+// the mdserver's time.  It's a very crude way of normalizing the
+// local clock.
+func (am *AutogitManager) commonTime(ctx context.Context) time.Time {
+	offset, haveOffset := am.config.MDServer().OffsetFromServerTime()
+	if !haveOffset {
+		am.log.CDebugf(ctx, "No offset, cannot use common time; "+
+			"falling back to local time")
+		return am.config.Clock().Now()
+	}
+	return am.config.Clock().Now().Add(-offset)
+}
+
+func (am *AutogitManager) canWorkOnRepo(
+	ctx context.Context, dstFS *libfs.FS, repo string) (
+	canWork bool, err error) {
+	am.log.CDebugf(ctx, "Checking if we can work on %s", repo)
+	defer func() {
+		am.deferLog.CDebugf(ctx, "Work check completed: canWork=%t, %+v",
+			canWork, err)
+	}()
+
+	// Take the lock for the dst repo while checking the work time.
+	lockFile, err := dstFS.Create(autogitLockName(repo))
```

Comment:
I suppose this doesn't have the same problem as in https://github.com/keybase/kbfs/pull/1446 where `Create` doesn't error if dst file exists, because we are relying on the `lockFile.Lock()` for synchronization.

Reply:
Yep, exactly.